### ISSUE TITLE
Orthographic Camera Support

### ISSUE THREAD

**Issue by AtaTrkgl** [CristianQiu/Unity-URP-Volumetric-Light#8]
I see that orthographic projection is currently not supported. Are there any plans to support it in the near future?

**Comment by CristianQiu** [comment]
Hi, I would like to say that I would have a look at it in the near future, but I think that it is currently unlikely. However, it would certainly be 'the' thing I would like to add as some users already asked for it. Sorry !

**Comment by OEN007** [comment]
I’m using it with an almost orthographic view and it looks great, but i guess you want it for 2d?

**Comment by AtaTrkgl** [comment]
@OEN007 Actually I'm working on a 3D game. The problem I had was, that even though the effect looks really well, when viewed from certain camera angles, artifacts start appearing.

**Comment by OEN007** [comment]
I see, what type of artifacts? Try bumping the resolution maybe. I think im running 2x the default

**Comment by AtaTrkgl** [comment]
If I center an object emmitting light, it's volume looks alright. But when I rotate the camera bit, the volume texture of that object shifts a bit, creating an artifact.

| Camera Centered | Camera Rotated a Bit |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/f532b868-32e9-4783-9d99-0ee3e1a19a1c) | ![image](https://github.com/user-attachments/assets/ef29be5b-53cf-42cf-bf36-d256df0de93a) |

**Comment by CristianQiu** [comment]
This package **does not support orthographic projection**. If you are using it, you will probably see the fog but it will not work as it should, and glitches are totally expected to happen.

If you are using perspective projection but with a narrower field of view (which I think it's what @OEN007 meant), to kind of get closer to an orthographic projection look, it should work fine. And finally, if you are seeing artifacts in perspective projection, this could be mainly to two reasons :

* Distance volume parameter is too high, and/or Max Steps parameter too low.
* Since blur is applied as a fullscreen effect when volumetric lighs are rendered to reduce noise, depth is considered to alter the blur weight. This means that depending on "what's behind" of the light volume, you may notice more artifacts. See the example below from Unity's URP Garden scene, where I tweaked the values to accentuate this inherent issue. There is nothing below the water, which does not write to the depth buffer, but nenuphars do, so more noise is expected to happen around that area.

![VolumetricLightNoise](https://github.com/user-attachments/assets/14455bf5-2cd5-4a96-ae3b-bc43a45d88be)

**Comment by OEN007** [comment]
> If you are using perspective projection but with a narrower field of view (which I think it's what @OEN007 meant), to kind of get closer to an orthographic projection look, it should work fine. And finally, if you are seeing artifacts in perspective projection, this could be mainly to two reasons :

Yes, I meant a narrow (low) field of view. Of course it will increase your rendering distance as you need to move the camera away as you zoom in but at small values but the volumetric effect works perfectly. You can adjust the near clip plane so you get enough depth resolution anyway.

**Comment by CristianQiu** [comment]
I have finally added orthographic camera support in v0.5.0.
Closing this, feel free to open new issues if you find anything ;)

**Comment by AtaTrkgl** [comment]
> I have finally added orthographic camera support in v0.5.0. Closing this, feel free to open new issues if you find anything ;)

Thank you so much, It works perfectly now! <3
